Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -150,9 +154,60 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           functionSymbol.Parameters.Add(ProcessFormal(scope, parameter));
         }
+        if (function.Result != null) {
+          functionSymbol.Parameters.Add(ProcessFormal(scope, function.Result));
+        }
+
+        Func<Expression, ScopeSymbol?> subP = (Expression e) => ProcessFunctionExpr(functionSymbol, e);
```

Comment:
Since you explicitly declare the type on the left-hand side of the assignment, it's not necessary to define the parameter type of the lambda expression.

Please do not use abbreviations.

Since `ProcessFunctionExpr` is only used here, you could declare it as a local method to capture the `functionSymbol` closure. However, I have generally a hard time following this code.